core/Types.h

```diff
@@ -995,7 +995,7 @@ struct DispatchArgs {
     const TypePtr &selfType;
     const TypeAndOrigins &fullType;
     const TypePtr &thisType;
-    const std::shared_ptr<const SendAndBlockLink> &block;
+    const SendAndBlockLink *const block;
```

Comment:
I guess from a high level, this change is fine, because the lifetime of a DispatchArgs was already "just as long as one `dispatchCall`, and is never meant to live longer than that?

One thing I was worried about at first was whether we ever "retained" a `DispactchArgs` (in the same way we retain a `DispatchResult` sometimes, e.g. in the LSP query match case) but I don't think we do that for `DispatchArgs`, so this probably doesn't change much about how people already have to reason about the lifetime of these objects.

Reply:
FWIW, retaining the `DispatchArgs` carries the same problems before and after this PR.  Technically, this PR makes things slightly safer, since before this PR, the stored reference for `block` would (always?) live on the stack and so would already be subject to stack-use-after-scope problems.  Whereas after, the pointer is still valid so long as somebody is holding the `shared_ptr` container.  🤣